### PR TITLE
Added support for extensive static analysis of web apps and sites

### PR DESCRIPTION
**Pull request by tforster**
Added a new sub command to Test:  `joy test staticWeb -u {url}`

This command performs an extensive analysis of the pages found at the specified url using [webhint.io](https://webhint.io/). Currently a small fraction of the possible tests are being run and we will need to determine either the default set of tests we feel are required, or, create our own set of pre-configured test suites that we can call from an additional cli argument.

FYI, prettier and eslint config still outstanding so let's not get hung up too much on style related issues that will be resolved by that task.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,6 +3,19 @@
   "version": "0.3.2",
   "description": "The Jake and trOY devopsy utility",
   "dependencies": {
```
Comment (webdivelement) [inline package.json]: Looks like it added a lot of dependencies/sub-dependencies to the overall project.

File: cli/plugins/test/index.js [hunk 2]
```diff
@@ -1,61 +1,86 @@
...
    * @param {*} joy : Instance of a Joy class c/w config.json, env and helper functions
-   * 
+   *
    */
   constructor(joy) {
     const prog = joy.prog;
     prog
       .command('test', 'Run various tests including a11y and static analysis')
-      .argument('<subcommand>', 'subcommand [a11y, static]', ['a11y', 'static'])
+      .argument('<subcommand>', 'subcommand [a11y, staticServer, staticWeb]', ['a11y', 'staticServer', 'staticWeb'])
```
Comment (webdivelement) [inline cli/plugins/test/index.js]: What are your thoughts on `camelCase` vs `lower-dash-separated` for the arguments?  Any pros/cons?

File: cli/plugins/test/index.js [hunk 3]
```diff
@@ -1,61 +1,86 @@
...
+      extends: ['web-recommended'],
+      formatters: ['stylish'],
+      parsers: ['css', 'html', 'javascript', 'manifest', 'package-json'],
+      browserslist: ['> 1%', 'last 2 versions']
+    };
 
+    const webhint = Analyzer.create(config);
+    const results = await webhint.analyze(options.urls[0]);
+
+    // TODO: Determine how we want to handle output (json to stdout? create excel file? display on screen? something else?)
```
Comment (webdivelement) [inline cli/plugins/test/index.js]: I think `stdout` is good for now, perhaps a flag or two down the road to export to a couple basic file types.

File: cli/plugins/test/index.js [hunk 4]
```diff
@@ -1,61 +1,86 @@
...
   }
 
+  /**
+   * Wraps [webhint](https://webhint.io/) to provide extensive analysis of front-end code
+   *
+   * @param {*} options : Additional CLI flags
+   * @param {*} logger :  Logger
+   */
+  async staticWeb(options, logger) {
+    const config = {
```
Comment (webdivelement) [inline cli/plugins/test/index.js]: I'll need to spend some time playing around with the options on some existing sites.  
We can also try using this on the templates we create to use with Joy.

**Comment by tforster** [comment]
I am being conscious of the deps but not so worried as it is a dev tool
used in a controlled environment by technically capable people. Would be
different if we had thousands of users relying on it as direct code. That
said I want to reduce some. Capital for instance, when we have time we can
replace it with our own code. Some of your questions about camel vs dash
were because of a caporal requirement.

File: package.json (hunk 1 above)
Comment (tforster) [reply package.json]: Most deps right now are for static site building as it's based on my gulpfile of the last few years. My separate static site generator project, back-burnered uses far less deps and I would expect to replace Joy code with that in a future update.

File: cli/plugins/test/index.js (hunk 2 above)
Comment (tforster) [reply cli/plugins/test/index.js]: I just discovered that Caporal will automatically convert dash-separated to dashSeparated when it passes the value in to code. So, I will start moving toward dash-separated as I touch each interface.

File: cli/plugins/test/index.js (hunk 3 above)
Comment (tforster) [reply cli/plugins/test/index.js]: https://github.com/tforster/joy/issues/11

**Comment by tforster** [comment]
@simpleimpulse Any comments or can you sign off on this one?

**Pull request merged**